src/System.Net.Sockets/src/System/Net/Sockets/ReceiveMessageOverlappedAsyncResult.Windows.cs

```diff
@@ -14,25 +14,36 @@ namespace System.Net.Sockets
 {
     internal unsafe sealed partial class ReceiveMessageOverlappedAsyncResult : BaseOverlappedAsyncResult
     {
-        private Interop.Winsock.WSAMsg* _message;
-        private WSABuffer* _wsaBuffer;
-        private byte[] _wsaBufferArray;
         private byte[] _controlBuffer;
-        internal byte[] _messageBuffer;
 
         private static readonly int s_controlDataSize = Marshal.SizeOf<Interop.Winsock.ControlData>();
         private static readonly int s_controlDataIPv6Size = Marshal.SizeOf<Interop.Winsock.ControlDataIPv6>();
-        private static readonly int s_wsaBufferSize = Marshal.SizeOf<WSABuffer>();
-        private static readonly int s_wsaMsgSize = Marshal.SizeOf<Interop.Winsock.WSAMsg>();
 
-        private IntPtr GetSocketAddressSizePtr()
+        internal unsafe byte* GetSocketAddressPtr()
         {
-            return Marshal.UnsafeAddrOfPinnedArrayElement(_socketAddress.Buffer, _socketAddress.GetAddressSizeOffset());
+            Debug.Assert(_socketAddress != null);
+            return (byte*)Marshal.UnsafeAddrOfPinnedArrayElement(_socketAddress.Buffer, 0);
+        }
+
+        internal unsafe int* GetSocketAddressSizePtr()
+        {
+            Debug.Assert(_socketAddress != null);
+            return (int*)Marshal.UnsafeAddrOfPinnedArrayElement(_socketAddress.Buffer, _socketAddress.GetAddressSizeOffset());
         }
 
         internal unsafe int GetSocketAddressSize()
         {
-            return *(int*)GetSocketAddressSizePtr();
+            return *(GetSocketAddressSizePtr());
+        }
```

Comment:
Looks like some of these functions are clones of earlier code.  Could they be defined once in some shared location, like as instance members of _socketAddress?